### PR TITLE
Fixed memory error for empty values in GraphIE.cpp AddConstantWithGraph

### PR DESCRIPTION
**Pull request by zesongw**
Added test case for Reshape 0D tensor and fixed the related bug.

### CODE REVIEW

File: src/webnn/tests/end2end/ReshapeTests.cpp [hunk 1]
```diff
@@ -24,11 +24,18 @@ class ReshapeTests : public WebnnTest {
...
-        std::vector<float> result(utils::SizeOfShape(expectedShape));
-        utils::Compute(graph, {{"a", inputData}}, {{"b", result}});
-        EXPECT_TRUE(utils::CheckValue(result, inputData));
+        if (newShape.size() == 0) {
+            const std::vector<float> inputData = {1};
+            std::vector<float> result(utils::SizeOfShape(expectedShape));
+            utils::Compute(graph, {{"a", inputData}}, {{"b", result}});
+            EXPECT_TRUE(utils::CheckValue(result, inputData));
+        } else {
+            const std::vector<float> inputData = {1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
```
Comment (fujunwei) [inline src/webnn/tests/end2end/ReshapeTests.cpp:33]: Merge L.29 - L.31 into L.35 - L.37
const std::vector<float> inputData = newShape.empty() ? {1} : {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12,13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24};

**Review by fujunwei — COMMENTED** [review]
Some comments, PTAL.

File: src/webnn/tests/end2end/ReshapeTests.cpp (hunk 1 above)
Comment (zesongw) [reply src/webnn/tests/end2end/ReshapeTests.cpp]: Merged

File: src/webnn/native/openvino/GraphIE.cpp [hunk 2]
```diff
@@ -126,6 +126,10 @@ namespace webnn::native::ie {
                 tensorDesc.dims.dims[i] = shape[i];
             }
             tensorDesc.dims.ranks = shape.size();
+            // Avoid memory error caused by empty vector's address. 
+            if (values.size() == 0) {
```
[the remark this answers is not in the document]
Comment (zesongw) [reply src/webnn/native/openvino/GraphIE.cpp]: I agree. It is better to handle this case in each operator.

**Comment by fujunwei** [comment]
@mingmingtasd Could you please take a look the memory leak in DirectMLX backend?

**Comment by mingmingtasd** [comment]
> @mingmingtasd Could you please take a look the memory leak in DirectMLX backend?

OK，I will. @fujunwei @zesongw

**Comment by mingmingtasd** [comment]
The root cause is you set an empty output shape which will let DMLX backend crash. Please try manually set newShape={1} above this line https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn/native/ops/Reshape.cpp#L29 @zesongw

**Comment by fujunwei** [comment]
Merge it, thanks.

**Pull request merged**